src/utils/trpc.ts

```diff
@@ -8,12 +8,23 @@ import { isDev } from '~/env/other';
 import { env } from '~/env/client.mjs';
 import { showErrorNotification } from '~/utils/notifications';
 
+type RequestHeaders = {
+  'x-client-date': string;
+  'x-client': string;
+  'x-client-version'?: string;
+  'x-fingerprint'?: string;
+};
+
 const url = '/api/trpc';
-const headers = {
+const headers: RequestHeaders = {
   'x-client-version': process.env.version,
   'x-client-date': Date.now().toString(),
   'x-client': 'web',
 };
+if (typeof window !== 'undefined') {
+  const fingerprint = window.localStorage.getItem('fingerprint');
+  if (fingerprint) headers['x-fingerprint'] = fingerprint;
+}
```

Comment:
Won't this only run on page load but the fingerprint gets added after?

Reply:
I believe this is the only place where it makes sense to set the fingerprint header globally and we can get the value from `localstorage`. So far the tests I've done always include the fingerprint when it's available